features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/Base/Header/EnvironmentMenu.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import Button from 'material-ui/Button';
+import Menu, {MenuItem} from 'material-ui/Menu';
+
+class EnvironmentMenu extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            openEnvironmentMenu: false
+        };
+    }
+
+    handleClickEnvironmentMenu = event => {
+        this.setState({openEnvironmentMenu: true, anchorElEnvironmentMenu: event.currentTarget});
+    };
+
+    handleRequestCloseEnvironmentMenu = () => {
+        this.setState({openEnvironmentMenu: false});
+        //TODO: [rnk] Temporary Fix: Reload the web page when environment changed
+        document.location.reload();
+    };
+
+    render() {
+        //Props list
+        const environments = this.props.environments;
+        const environmentLabel = this.props.environmentLabel;
+        const handleEnvironmentChange = (event) => {
```

Comment:
AFAIK there is a small performance impact on creating arrow function when always rendering the element, shall we move this function to class level

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/4890/commits/57e9929cdc68c5b4447681c94a793b86b0390380